Add pre commit checks

Also, remove `git fetch --unshallow || true`.

Seems like an error with that as well.

```
fatal: --unshallow on a complete repository does not make sense
❌ Merge commit found!
Error: Process completed with exit code 1.
```

Try running the workflow in any private repo of your own before making PR.

--- a/.github/workflows/pr-checks.yaml
+++ b/.github/workflows/pr-checks.yaml
@@ -21,6 +21,7 @@ jobs:
       - name: Calculate changed lines
         id: changed-lines
         run: |
+          set +e
           ADDED=$(git diff --shortstat ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} | grep -o '[0-9]\+ insertions' | grep -o '[0-9]\+')
           DELETED=$(git diff --shortstat ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} | grep -o '[0-9]\+ deletions' | grep -o '[0-9]\+')
           ADDED=${ADDED:-0}

--- a/.github/workflows/pr-checks.yaml
+++ b/.github/workflows/pr-checks.yaml
@@ -1,7 +1,7 @@
 name: PR Quality Checks
 
 on:
-  pull_request:
+  pull_request_target:
     types: [opened, synchronize, reopened, ready_for_review]
     branches: [staging, prod]
 

